docs/v1/api.md

```diff
@@ -0,0 +1,333 @@
+controllers in a cluster. We believe this allowance encourages flexibility in
+a cluster's service catalog.
+
+# Kubernetes Resources
+
+Given these inspirations, the following resources will be added to the
+Kubernetes API:
+
+- `ServiceClass`
+  - Describes a _kind_ of service. This resource is semantically similar to how
+    a `StorageClass` defines a _kind_ of persistent volume, but we expect that,
+    in practice, the kinds of services represented here will be more coarse-
+    grained than the kinds of volumes represented in `StorageClass`es
+  - References a service ID and plan ID, which is managed and understood by a
+    service controller (see below)
+  - Created by an operator or tool
+- `ServiceInstance`
+  - An instance of a `ServiceClass` -- an in-cluster representation of a
+    provisioned service
+  - Can be statically provisioned by an operator, to enable multiple consumers
+    to bind to it (enables multi-tenancy)
+  - Can be dynamically provisioned and bound to by a consumer, to guarantee an
+    app’s exclusive use of the resource (guarantees single-tenancy)
+- `ServiceClassClaim`
+  - Created by a consumer to provision an instance of a `ServiceClass`, then
+    bind to the provisioned. Enables dedicated, single-tenant usage of a
+    provisioned resource
+- `ServiceClassBinding`
+  - Created in response to a successfully executed `ServiceClassClaim`
+  - References credentials created as a result of the bind operation
+  - Holds a record of an application that created a `ServiceClassClaim` which
+    was successfully executed
+- `ServiceInstanceClaim`
+  - Created by a consumer to bind to an existing `ServiceInstance` by name.
+    Enables multi-tenant usage of a provisioned resource
+- `ServiceInstanceBinding`
+  - Created in response to a successfully executed `ServiceInstanceClaim`
+  - References credentials created as a result of the bind operation
+  - Holds a record of an application that created a `ServiceInstance` which
+    was successfully executed
+
+
+The below diagram shows the interactions between these resources:
+
+![Resource Interactions](./img/partial-flow.png)
+
+# Example Resources
+
+## `ServiceClass`
+
+```yaml
+kind: ServiceClass
+apiVersion: service-catalog.k8s.io/alpha
+metadata:
+  name: postgres-small
+spec:
+  serviceID: c5a0ad8e-b57b…
+  planID: c68cd6b1-be9f…
+```
+
+## `ServiceInstance`
+
+```yaml
+kind: ServiceInstance
+apiVersion: service-catalog.k8s.io/alpha
+metadata:
+  name: zip-code-db
+spec:
+  serviceClass: postgres-small
+```
+
+## Single Tenant / Dedicated Resources
+
+The `ServiceClassClaim` and `ServiceClassBinding` resources are created
+to give an application single-tenant, dedicated access to a backing resource.
+
+The developer should create a `ServiceClassClaim` resource, and the system
+should do the `provision` operation, then `bind` immediately thereafter.
+
+### ServiceClassClaim
+
+```yaml
+kind: ServiceClassClaim
```

Comment:
Why not just use a ServiceInstanceClaim with an optional serviceClass field instead? The way Pod-Node and PVC-PV work is that the claims specify the properties of the resource that will satisfy their requirements (including class), and then a binding operation is used to update the claim with the name of the resource selected. The name of the resource can also be provided up front, in which case automatic selection is bypassed.